provisioning/device/samples/readme.md

```diff
@@ -0,0 +1,101 @@
+
+2. Individual Enrollment
+In this case, each device certificate gets its own certificate uploaded to the service.
+_Note:_ the device certificate can also be a signing certificate of the actual certificate used to authenticate with the service during provisioning.
+
+An example of specifying the authentication X509Certificate using a PKCS12 PFX password-protected file:
+
+```C#
+    using (var certificate = new X509Certificate2(s_certificateFileName, certificatePassword))
+    using (var security = new SecurityClientX509(certificate))
+    {
+        //...
+    }
+```
+
+The SDK provides an extension model [SecurityClientHsmX509](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/shared/Microsoft.Azure.Devices.Shared/SecurityClientHsmX509.cs) that allows hardware vendors to implement custom Hardware Security Modules that store the device certificates. On Windows, PKCS11 HSM devices are supported through the [Certificate Store](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/certificate-stores).
+
+An example of implementation for this extension module is the [SecurityClientX509](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/shared/Microsoft.Azure.Devices.Shared/SecurityClientX509.cs) class.
+
+### Provisioning devices using TPM based attestation
+
+In the case of TPM attestation, a RegistrationID must be supplied by the application.
+
+The TPM attestation supports only Individual Enrollments. The [Endorsement Key](https://technet.microsoft.com/en-us/library/cc770443(v=ws.11).aspx) (EK) must be supplied to the service. During provisioning, the [Storage Root Key](https://technet.microsoft.com/en-us/library/cc753560(v=ws.11).aspx) (SRK) will also be used to ensure that TPM ownership changes result in devices provisioned to the correct owner.
+
+
+```C#
+    using (var security = new SecurityClientTpmSimulator(RegistrationId))
+    {
+        //...
+    }
+```
+
+The SDK provides an extension model [SecurityClientHsmTpm](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/shared/Microsoft.Azure.Devices.Shared/SecurityClientHsmTpm.cs) that allows hardware vendors to implement custom TPM v2.0 Hardware Security Modules.
+
+The samples use a TPMv2.0 simulator that uses a loopback TCP connection for communication. This is provided for demonstration purposes only and does not provide any security.
+
+## List of samples
+
+- [Provisioning Devices with X509 certificate attestation](ProvisioningDeviceClientX509)
+- [Provisioning Devices with TPM attestation](ProvisioningDeviceClientTpm)
+
+### How to run the samples
+
+_Preview only:_ Running the samples requires building from sources. This will not be necessary once NuGet packages are released.
+
+1. Prepare your development environment. Follow the instructions at https://github.com/Azure/azure-iot-sdk-csharp/blob/master/device/doc/devbox_setup.md
+
+2. Build the environment. In the root of your clone, type:
+
+```build -clean -wip_provisioning```
```

Comment:
azureiot-sdk-csharp or the root of the sample folder?

Reply:
When I run this command from azure-iot-sdk-csharp, I get 

.\build.ps1 : File C:\Users\yizhon\azure-iot-sdk-csharp\build.ps1 cannot be loaded because running scripts is disabled
on this system. For more information, see about_Execution_Policies at http://go.microsoft.com/fwlink/?LinkID=135170.
At line:1 char:1
+ .\build.ps1 -clean -wip_provisioning ; exit $LASTEXITCODE
+ ~~~~~~~~~~~
    + CategoryInfo          : SecurityError: (:) [], PSSecurityException
    + FullyQualifiedErrorId : UnauthorizedAccess

When I run it from azure-iot-sdk-csharp\provisioning\device\samples\ProvisioningDeviceClientTpm>build -clean -wip_provisioning
'build' is not recognized as an internal or external command,
operable program or batch file.

Reply:
The root of the clone. 

Your system is blocking Powershell script execution. This is a common case. Developers usually lower the security requirement by running the following in an Admin Powershell script:

`Set-ExecutionPolicy RemoteSigned`

Since this comes with a security risk and the error correctly informs the user and gives out the link on what that means, I prefer not documenting how to disable the security feature.

Reply:
Maybe we can say, Powershell script execution must be enabled on this device

Reply:
Sounds good. I can also add a link to the docs on how to do that.